docs/sections/contributor_guide/developer_setup.rst

```diff
@@ -1,5 +1,183 @@
+.. include:: ../external_links.rst
+
 ***************
 Developer Setup
 ***************
 
-TBD
+Creating a development shell
+============================
+
+To create an interactive development ``bash`` shell:
+
+This recipe uses the ``aarch64`` (64-bit ARM) Miniforge for Linux, and installs into ``$HOME/conda``. Adjust as necessary for your target system.
+
+1. Download, install, and activate the latest `Miniforge3`_ for your system. If an existing conda (Miniforge, Miniconda, Anaconda, etc.) installation is available and writable, you may activate that and skip this step and continue on to the next.
+
+   .. code:: sh
+   
+     wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-aarch64.sh
+     bash Miniforge3-Linux-aarch64.sh -bfp ~/conda
+     rm Miniforge3-Linux-aarch64.sh
+     source ~/conda/etc/profile.d/conda.sh
+     conda activate
+
+2. Install the `condev`_ package into the base environment.
+
+   .. code:: sh
+
+     conda install -y -c maddenp condev
+
+3. In a clone of the `workflow-tools repository`_, create the development shell.
+
+
+   .. code:: sh
+
+     cd /to/your/workflow-tools/clone
+     make devshell
+
+
+.. _paragraph: 
+
+If the above is successful, you will be in a ``workflow-tools``
+development shell. See below for usage information. You may exit the
+shell with ``exit`` or ``ctrl-d``.
+
+Future ``make devshell`` invocations will be almost instantaneous, as
+the underlying virtual environment will already exist. In general, all
+source code changes will be immediately live in the development shell,
+subject to execution, test, etc. But some changes – especially to the
+contents of the ``recipe/`` directory, or to the ``src/setup.py`` module
+– may require recreation of the development shell. If you know this is
+needed, or when in doubt: 
+
+    1. Exit the development shell, run ``conda env remove -n DEV-uwtools`` to remove the old environment.
+    2. Run ``make devshell`` to recreate it.
+
+If your development shell misses any functionality you’re used to in
+your main shell, you can create a ``~/.condevrc`` file, which will be
+sourced by ``make devshell``. When in doubt, you might:
+
+   .. code::
+
+     cat <<EOF >~/.condevrc
+     source ~/.bashrc
+     EOF
+
+
+Using a development shell
+=========================
+
+In an active development shell, the following ``make`` targets are
+available and act on all ``.py`` files under ``src/``:
+
+
++---------------------+------------------------------------------------------------+
+| Command             |  Description                                               |
++=====================+============================================================+
+| ``make format``     | Format with |black|_ and |isort|_                          |
```

Comment:
was able to add links to `black` and `isort`. There isn't a good way to add a hyperlink inside a code section. So far from what I found from other resources this was the best work around. However, when you look at the actual webpage for our documentation you will see that `black` and `isort` do not get highlighted blue and only when you have your cursor over them do you realize it is a hyperlink.

Some work still needs to be done on this. Hoping to find a solution to incorporate both.

Reply:
I think I'd be fine leaving it as just a link (not also a code section) since one of the perks of making it a code section is to highlight it as a package. Having it just live here as a URL is important because it both highlights the word and lets the reader know that they can find more information about what black is/does.